runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/GrpcWindmillServer.java

```diff
@@ -969,6 +987,76 @@ protected void startThrottleTimer() {
       getWorkThrottleTimer.start();
     }
 
+    private class GetWorkTimingInfosTracker {
+      private final Map<State, Duration> getWorkStreamLatencies;
+
+      public GetWorkTimingInfosTracker() {
+        this.getWorkStreamLatencies = new EnumMap<>(State.class);
+      }
+
+      public void addTimingInfo(Collection<GetWorkStreamTimingInfo> infos) {
+        // We want to record duration for each stage and also be reflective on total work item
+        // processing time. It can be tricky because timings of different
+        // StreamingGetWorkResponseChunks can be interleaved. Current strategy is to record the
+        // maximum duration in each stage across different chunks, this will allow us to identify
+        // the slow stage, but note the sum duration of each slowest stages may be larger than the
+        // duration from first chunk creation to last chunk reception by user worker.
```

Comment:
I think that we might want to scale them perhaps?
I think that it will be confusing to compare these latencies to latencies measured on the user worker which match real-time elapsing otherwise.

Reply:
Do you mind elaborate on how to scale them, do you mean apply a multiplier?
From testing it looks like the GetWork creation and transmission are on the magnitude of  0 ~ 200 ms for a work item (unless there is significant delay). They are likely incomparable to the latencies measured on the user worker anyways which is a few seconds to a few minutes most likely.

Reply:
I was thinking to scale the times to transmit to user worker such that those portions of the latency attribution match the time it takes from generating work to assembling it on the user worker.

So the transmit time elapsed would be something like [GET_WORK_CREATION_END time, last chunk arrived in user worker].
Then we can scale the components of latency for transmitting (ie GET_WORK_IN_TRANSIT_TO_USER_WORKER and GET_WORK_IN_TRANSIT_TO_DISPATCHER) so that their sum equals the transmit time.

I think it's worthwhile as we are looking into low-latency processing because it isn't necessarily guaranteed that user worker processing will take longer and it's confusing if the total of all the latencies could be larger than the actual total time to process.

Reply:
Done. I think it would make sense to scale the work creation time as well so the total elapsed time would be [GET_WORK_CREATION_START, last chunk arrived in user worker].